### PR TITLE
feat: add npm_translate_lock(update_pnpm_lock_node_toolchain_prefix) to specify node toolchain

### PR DESCRIPTION
**Pull request by jbedard**

### CODE REVIEW

**Review by alexeagle — COMMENTED** [review]
This new attribute is only meant to be used when we are doing `update_pnpm_lock` so 
- add a validation so we `fail` if this is set but `update_pnpm_lock` isn't
- (maybe) name the attribute so the pairing is clear, if we expect we'll never want the node toolchain in `npm_translate_lock` in any other case

File: npm/private/npm_translate_lock_state.bzl [hunk 1]
```diff
@@ -150,7 +150,7 @@ def _init_pnpm_labels(label_store, rctx):
     #
     # TODO: Try to understand this better and see if we can go back to using
```
Comment (alexeagle) [inline npm/private/npm_translate_lock_state.bzl:151]: since we're here and studying this, could we remove the TODO as well?

**Comment by jbedard** [comment]
I've added the `update_pnpm_lock_` prefix to the attribute name. I'm not sure if it's worth adding validation though which would require more logic then just `= "nodejs"` as a default param for it. Do you think it's worth it?

File: npm/private/npm_translate_lock_state.bzl (hunk 1 above)
Comment (jbedard) [reply npm/private/npm_translate_lock_state.bzl:151]: I think that's out of scope here and I'd rather not put the time in atm.

**Review by alexeagle — APPROVED** [review]
sure, looks good enough to me, thanks

**Pull request merged**